### PR TITLE
classes: bundle: Allow parsing var flags offset, hooks and adaptive

### PR DESCRIPTION
**Pull request by mschwan-phytec**
Allow for BitBake variable parsing inside the variable flags "offset", "hooks" and "adaptive". To enable this, use d.getVarFlag() instead of the builtin get() method.

### CODE REVIEW

File: classes/bundle.bbclass [hunk 1]
```diff
@@ -281,9 +281,9 @@ def write_manifest(d):
         if slotflags and 'hooks' in slotflags:
             if not have_hookfile:
                 bb.warn("A hook is defined for slot %s, but RAUC_BUNDLE_HOOKS[file] is not defined" % slot)
-            manifest.write("hooks=%s\n" % slotflags.get('hooks'))
+            manifest.write("hooks=%s\n" % d.getVarFlag('RAUC_SLOT_%' % slot, 'hooks'))
```
Comment (ejoerns) [inline classes/bundle.bbclass:284]:
```suggestion
            manifest.write("hooks=%s\n" % d.getVarFlag('RAUC_SLOT_%s' % slot, 'hooks'))
```

File: classes/bundle.bbclass [hunk 2]
```diff
@@ -281,9 +281,9 @@ def write_manifest(d):
...
         if slotflags and 'adaptive' in slotflags:
-            manifest.write("adaptive=%s\n" % slotflags.get('adaptive'))
+            manifest.write("adaptive=%s\n" % d.getVarFlag('RAUC_SLOT_%' % slot, 'adaptive'))
```
Comment (ejoerns) [inline classes/bundle.bbclass:286]:
```suggestion
            manifest.write("adaptive=%s\n" % d.getVarFlag('RAUC_SLOT_%s' % slot, 'adaptive'))
```

**Review by ejoerns — CHANGES_REQUESTED** [review]
Good catch about the parsing handling.

But pretty sure this was only tested for `'offset'` :wink:

**Comment by mschwan-phytec** [comment]
> Good catch about the parsing handling.
> 
> But pretty sure this was only tested for `'offset'` :wink:

Whoops! :grimacing: Fixed.

**Review by ejoerns — APPROVED** [review]

**Comment by mschwan-phytec** [comment]
Can we backport this to kirkstone? As adaptive updates are available in kirkstone, too, it would fix the parsing there.

**Comment by ejoerns** [comment]
cherry-picked to `kirkstone`

**Pull request merged**